src/pages/Account/BankDetails.jsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from 'react'
+import { useSelector, shallowEqual, useDispatch } from 'react-redux'
+import * as Yup from 'yup'
+
+
+import { disableButton } from '../../store/reducers/buttonState'
+import { Form, FormField, FormSelect, SubmitButton } from '../../components/Form'
+import { updateUserBank } from '../../store/reducers/user'
+import { BANKS, getIndexOfK } from '../../utils/helperFunctions'
+
+
+
+const validationSchema = Yup.object().shape({
+    bankName: Yup.string().required().label("Bank Name"),
+    accountName: Yup.string().required().min(3).label("Account Name"),
+    accountNumber: Yup.string().required().min(10).max(10).label("Account Number")
+})
+
+
+const BankDetails = ({ user }) => {
+
+    const {bankName, accountName, accountNumber} = user
+
+    const[editable, toggleEdit] = useState(false)
+
+
+
+    const { buttonDisable } = useSelector(state => ({
+        buttonDisable : state.buttonState.buttonDisable
+    }), shallowEqual)
+
+
+
+    const dispatch = useDispatch()
+
+
+
+    const submitForm = ({ accountName, accountNumber, bankName }) => {
+
+        const userData = {
+            bankName,
+            accountName,
+            accountNumber
+        }
+
+        dispatch(disableButton())
+        dispatch(updateUserBank(userData))
+
+    }
+
+
+    
+
+    return(
+        <>
+            <h4>Banking Details</h4>
+
+
+                <Form
+                        initialValues={{ 
+                            bankName,
+                            accountName,
+                            accountNumber
+                        }}
+                        onSubmit={submitForm}
+                        validationSchema={validationSchema}
+                    >
+
+                        
+                        <FormSelect 
+                            name="bankName"
+                            options={BANKS}
+                            index={getIndexOfK(BANKS, bankName)[0]}
+                            disabled={!editable}
+                        />
+
+                        <FormField 
+                            type="text"
+                            name="accountName"
+                            placeholder="Name of Account"
+                            disabled={!editable}
+                        />
+
+                        <FormField 
+                            type="text"
+                            name="accountNumber"
+                            placeholder="Bank Account Number"
+                            disabled={!editable}
+                        />
```

Comment:
I feel instead of letting the users input the account name, bank name and account number they should just enter the account name and bank name then we verify the account and show them the name of that account if the account exists otherwise we show an error message, so as to be sure that the user doesn't make mistake by typing invalid details.

Reply:
Sure...that is a splendid idea. Do you know any plugin we could use?

Reply:
Yh, I've tried something like this with the paystack API
We need two APIs here, one to get all bank names and their respective codes, so instead of hard coding it we fetch it from the API then we use the other API endpoint which we pass the account number and bank code and it returns account number, bank name, and account name


You can check the API reference here: https://paystack.com/docs/identity-verification/resolve-account-number

Reply:
Or better still we ask them for the full details then we use this API to verify and if we get a response other than success we let the user know that they supplied data that is not valid.

Reply:
https://moneywave.flutterwave.com/api

Reply:
https://paystack.com/docs/api/#miscellaneous-bank
We can get list of bank codes using this API, but I'm thinking since we know that this data won't change we can just remove the code and name and store it in a file instead of having to make requests for this and since what is returned is more than just code and name it's expected to be slower.

Reply:
I'll figure this out with a simpler api